# Probability Review

http://onlinestatbook.com/2/probability/probability_intro.html

In many events, there are no definite outcomes so the outcome can't be predicted with total certainty. What we can say is how likely the outcomes are to happen, using the idea of probability? Inferential statistics is built on the foundation of probability theory and has been remarkably successful in guiding opinion about the conclusions to be drawn from data. 

One conception of probability is drawn from the idea of symmetrical outcomes. For example, the 2 possible outcomes of tossing a fair coin seem not to be distinguishable in any way that affects which side will land up or down. Therefore, the probability of heads is taken to be 1/2 (same as tails). In general, if there are N symmetrical outcomes, the probability of any given one of them occurring is taken to be 1/N. Thus, if a six-sided die is rolled, the probability of any one of the six sides coming up is 1/6. 

If the oil prices is traded at $100 a barrel 70% of the time in the last year then the probability of it being around $100 would be 70% higher for the next year. This is a common conclusion but could be unreasonable if there is more data available to decide whether it will be trading at $100 a barrel tomorrow. For example, if the supply increased in the last day then the price would fall. We should consider only the days in last year where oil production matches. Even this information is not enough since decrease in oil prices depends on consumption too (The prices will fall if consumption is low). So, we should consider only the prior occurrences of matching oil production and similar consumption. As we keep considering more factors affecting the outcomes, you can see that the sample of prior cases will soon be reduced to the empty set. 

In some cases, probability should be though of as subjective. Questions such as "What is the probability the Starbucks coffee is better than McDonalds coffee" can't be answer using summetry or frequency approaches to probability. Assigning a probability of 0.9 for example to this event reflects the speaker's personal opinion. Such an approach to probability, however, seems to lose the objective content of the idea of chance; probability becomes mere opinion. 

### Sample Spaces

For a random experiment E, the set of all possible outcomes of E is called the sample space and is denoted by the letter S. 
For a coin-toss experiment, S would be the results “Head” and “Tail”, which we may represent by $S = {H, T}$. Formally, the performance of a random experiment is the unpredictable selection of an outcome in S.

The R package `prob` has all the functions to find probability of different basic events. 
A sample space is (usually) represented by a data frame.
Each row of the data frame corresponds to an outcome of the experiment.

Consider the random experiment of tossing a coin.
The outcomes are H and T. 
We can set up the sample space quickly with the tosscoin function:

In [3]:
library(prob)

Loading required package: combinat

Attaching package: 'combinat'

The following object is masked from 'package:utils':

    combn


Attaching package: 'prob'

The following objects are masked from 'package:base':

    intersect, setdiff, union



In [4]:
tosscoin(1)

toss1
1 H    
2 T

The number 1 tells tosscoin that we only want to toss the coin once. We could toss it more times, like tosscoin(3), to get the output below...

In [5]:
tosscoin(3)

toss1 toss2 toss3
1 H     H     H    
2 T     H     H    
3 H     T     H    
4 T     T     H    
5 H     H     T    
6 T     H     T    
7 H     T     T    
8 T     T     T

## Events

An event A is merely a collection of outcomes, or in other words, a subset of the sample space. 
After the performance of a random experiment E we say that the event A occurred if the experiment’s outcome was A. We say that a bunch of events-- A1, A2, A3, . . . -- 
are mutually exclusive or disjoint if Ai $\cap$ Aj = $\emptyset$; for any distinct pair Ai $\ne$ Aj. 
For instance, in the coin-toss experiment the events A = {Heads} and B = {Tails} would be mutually exclusive.

In [6]:
subset(rolldie(3), X1 + X2 + X3 > 16)

X1 X2 X3
180 6  6  5 
210 6  5  6 
215 5  6  6 
216 6  6  6

Since the die is rolled 3 times, total number of possible outcomes are $6^3=216$. There are 4 outcomes as listed above where the sum of numbers appeared on dice in three rolls is greater than 16. 

#### Functions for Finding Subsets

**The `%in%` function**

The function %in% helps to learn whether each value of one vector lies somewhere inside another vector.

In [7]:
x <- 1:10
y <- 8:12
y %in% x

[1]  TRUE  TRUE  TRUE FALSE FALSE

Notice that the returned value is a vector of length 5 which tests whether each element of `y` is in `x`, in turn.

----

**The `isin` function**

It is more common to want to know whether the whole vector y is in x. We can do this with the isin function.

In [8]:
isin(x, y)

[1] FALSE

Note that there is an optional argument ordered, which tests whether the elements of y appear in x in the order in which they are appear in y. The output is...

In [9]:
isin(x, c(3, 4, 5), ordered = TRUE)

[1] TRUE

In [10]:
isin(x, c(3, 5, 4), ordered = TRUE)

[1] FALSE

### Properties of Probability

For any events A and B,

I. $P(A^c) = 1 - P(A)$.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Proof: Since $A \cup A^c = S$ and $A \cap A^c = \emptyset$, we have

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $1 = P(S) = P(A \cup A^c) = P(A) + P(A^c)$.

----

II. P($\varnothing$) = 0.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Proof: Note that $\varnothing = S^c$, and use Property II.
    
----

III. If $A \subset B$ , then $P(A) \le P(B)$.
    
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Proof: Write $B = A \cup (B \cap A^c)$, and notice that $A \cap (B \cap A^c)$ = $\varnothing$; thus
    
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $P(B) = P(A \cup (B \cap A^c)) = P(A) + P (B \cap A^c) \ge P(A);   
   
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; since $P (B \cap A^c) \ge 0$

----

IV. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $0 \le P(A) \le 1$

----

V. The General Addition Rule.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $P(A \cup B) = P(A) + P(B) + P(A \cap B)$

----

VI. The Theorem of Total Probability. Let $B_1, B_2, . . . , B_n$ be mutually exclusive and exhaustive.

Then $P(A) = P(A \cap B_1) + P(A \cap B_2) + ... + P(A \cap B_n)$

### Counting Methods

There are different counting methods. 
The equally-likely model assigns equal probability to all elementary outcomes of the experiment and is a convenient and popular way to analyze random experiments. 
When the equally likely model applies, finding the probability of an event `A` amounts to nothing more than counting the number of outcomes that `A` contains (together with the number of events in S). 
Hence, to be a master of probability one must be skilled at counting outcomes in events of all kinds.

**The Multiplication Principle**: 
Suppose that an experiment is composed of two successive steps. 
Further suppose that the first step may be performed in $n_1$ distinct ways while the second step may be performed in $n_2$ distinct ways. 
Then the experiment may be performed in $n_1 n_2$ distinct ways.

More generally, if the experiment is composed of `k` successive steps which may be performed in $n_1, n_2, . . . , n_k$ distinct ways, respectively, then the experiment may be performed in $n_1 n_2 ... n_k$ distinct ways.



----

**Example:** We would like to order a pizza. 
It will be sure to have cheese (and marinara sauce) but we may elect to add one or more of the following five (5) available toppings: 

        pepperoni, sausage, anchovies, olives, and green peppers.

How many distinct pizzas are possible?


**Answer:** There are many ways to approach the problem, but the quickest avenue employs the Multiplication Principle directly.
We will separate the action of ordering the pizza into a series of stages. At the first stage, we will decide whether or not to include pepperoni on the pizza (two possibilities). 
At the next stage, we will decide whether or not to include sausage on the pizza (again, two possibilities). 
We will continue in this fashion until at last we will decide whether or not to include green peppers on the pizza.

At each stage we will have had two options, or ways, to select a pizza to be made. 
The Multiplication Principle says that we should multiply the 2’s to find the total number of possible pizzas:
$$2 * 2 * 2 * 2 * 2 = 2^5 = 32$$

### Ordered Samples

The number of ways in which one may select an ordered sample of k subjects from a population that has n distinguishable members is
* $n^k$ if sampling is done with replacement,
* n(n - 1)(n - 2) ... (n - k + 1) if sampling is done without replacement.

Recall from math the notation for factorials:


$1! = 1, $

$2! = 2 . 1 = 2,$

$3! = 3 . 2 . 1 = 6,$

$.$

$.$

$.$

$n! = n(n-1)(n-1)...3.2.1$


**Example:** We rent five movies to watch over the span of two nights. 
We wish to watch 3 movies on the first night. 
How many distinct sequences of 3 movies could we possibly watch?

**Answer:** $5 * 4 * 3 = 60$.

### Unordered Samples

The number of ways in which one may select an unordered sample of `k` subjects from a population that has `n` distinguishable members is:

 * $\frac{(n - 1 + k)!}{[(n - 1)!k!]}$ if sampling is done with replacement, 

 * $\frac{n!}{[k!(n - k)!]}$ if sampling is done without replacement.

The quantity $\frac{n!}{[k!(n-k)!]}$ is called a binomial coefficient and plays a special role in mathematics. 
It is denoted  

$$\binom nk  = \frac{n!}{k!(n-k)!}$$


**Example:** 
You rent five movies to watch over the span of two nights, 
but only wish to watch 3 movies the first night. 
Your friend, Fred, wishes to borrow some movies to watch at his house on the first night. 
You owe Fred a favor, and allow him to select 2 movies from the set of 5. How many choices does Fred have? 

**Answer:** $\binom 52  = 10$


**Example: ** 
Let our urn simply contain three balls, labeled 1, 2, and 3, respectively. 
We are going to take a sample of size 2 from the urn.

**Answer: ** 
The `prob` package accomplishes sampling from urns with the urnsamples function, which has arguments x, size, replace, and ordered. 
The argument `x` represents the urn from which sampling is to be done. 
The `size` argument tells how large the sample will be. 
The ordered and replace arguments are logical and specify how sampling will be performed. 
We will discuss each in turn. 

If sampling is with replacement, then we can get any outcome 1, 2, or 3 on any draw. 
Further, by "ordered" we mean that we shall keep track of the order of the draws that we observe. 
We took a sample of size 2 from an urn with three distinguishable elements.

In [12]:
nsamp(n = 3, k = 2, replace = TRUE, ordered = TRUE)

[1] 9

The total number of outcomes for above event are 9. 